verilog/analysis/flow_tree.h

```diff
@@ -0,0 +1,133 @@
+// Copyright 2017-2022 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef VERIBLE_VERILOG_FLOW_TREE_H_
+#define VERIBLE_VERILOG_FLOW_TREE_H_
+
+#include <bitset>
+#include <map>
+#include <string>
+#include <vector>
+
+#include "absl/status/status.h"
+#include "common/lexer/token_stream_adapter.h"
+#include "verilog/parser/verilog_token_enum.h"
+
+namespace verilog {
+
+// FlowTree class builds the control flow graph of a tokenized System-Verilog
+// source code. Furthermore, enabling doing the following queries on the graph:
+// 1) Generating all the possible variants (provided via a callback function).
+// 2) TODO(karimtera): uniquely identify a variant with a bitset.
+class FlowTree {
+ public:
+  using BitSet = std::bitset<128>;
+  using TokenSequenceConstIterator = verible::TokenSequence::const_iterator;
+
+  struct ConditionalBlock {
```

Comment:
The name `ConditionalBlock` and then seeing the fields it has does not necessarily connect, mostly because of the naming of the fields inside (suggestion below). But also, like any class or struct, this of course wants a description.

The naming of the fields is odd. Typically we call something an `iterator` if it is something you'd want to use to advance through some sequence of things. So while these are an iterator type, here, we're mostly using them to remember a position; maybe call them '`pos` or `postition`, `location` or something of that sort ?
e.g. `else_location`.

There needs to be a description of where the iterators point to when they are not pointing anywhere; possibly the `end()` of some token sequence, but it is not entirely clear what that would be. Maybe a short description that this `ConditionalBlock` struct is always used in conjunction with some corresponding `TokenSequence` and that all of the fields in here are pointing inside there.

Also, it seems ifdef and ifndef could be exclusive, maybe that requires some documentation or a different code to make it impossible to mis-use. Maybe only one `if_location` that either points to either an `ifdef` if `ifndef`  and a boolean indicating if this is a positive condition ?